### PR TITLE
Add algebra

### PR DESCRIPTION
**Pull request by fibo**
Hi, I added my algebra package.

Currently I am adding more documentation, tests and cleaning the code to prepare for a 1.0 version. The API is almost stable right now.

The algebra package let also define custom groups and rings, I did not added it in the description to keep it shorter.

Complex and Quaternion are implemented as composition algebras using the Cayley-Dickson operator over Reals, so also Octonion are implemented as well as Complexes over a custom field (for example using strings instead of numbers, if you find operators that respect the ring definition laws).

I hope algebra can enter the awesome node list.

Thanks,
Gianluca.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Please read and adhere to the [contributing guidelines](../blob/master/contributing.md).

**Comment by fibo** [comment]
I read it, I guess I missed

> Additions should be added to the bottom of the relevant category.

If yes, I moved *algebra* package to the bottom of its category. Sorry, I added on top to follow alphabetical order.

Please tell me if there is any other modification I should do to adhere.

File: readme.md [hunk 1]
```diff
@@ -443,7 +443,7 @@
 - [mathjs](https://github.com/josdejong/mathjs) - An extensive math library.
 - [math-sum](https://github.com/sindresorhus/math-sum) - Sum numbers.
 - [math-clamp](https://github.com/sindresorhus/math-clamp) - Clamp a number.
-
+- [algebra](https://github.com/fibo/algebra) - Vectors, Matrices, Tensors. Real, Complex, Quaternion.
```
Comment (sindresorhus) [inline readme.md]: Needs a better description than that.

**Comment by fibo** [comment]
Ahah, it is the same description I put in the package.json; what about

> Tensors over any field, including reals of course.

**Comment by sindresorhus** [comment]
Why are you suddenly linking to a different module?

**Comment by fibo** [comment]
It is a dependency of algebra package, it has a more concise description.
I would like to add algebra, but, I need help to describe it in a short way.
Let me try again.
Hope this description is ok.

**Comment by fibo** [comment]
Awesome thanks!
By the way, thanks to awesome-nodejs I got a collaboration for [algebra-group](https://github.com/fibo/algebra-group) which stimulated me to improve documentation.

Also, *algebra* download trend seems they are growing, see [npm stats](http://npm-stat.com/charts.html?package=algebra&author=&from=&to=).
Aside that spike that was one I released few days changing  the description, cause thanks to your critic I realized that old description was ok for a mathematician, not for all other people.

Now the package description is:

**algebra**

> means completeness and balancing, from the Arabic word الجبر

Thanks for your awesome list!
